src/components/calculator.js

```diff
@@ -1,57 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './calculator.css';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
 
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-
-    this.handleEvent = this.handleEvent.bind(this);
-  }
-
-  handleEvent = (buttonName) => {
-    this.setState((state) => calculate(state, buttonName));
+  const handleEvent = (e) => {
+    setState({ ...state, ...calculate(state, e.target.name) });
   };
```

Comment:
- [ ] When an operator button is clicked multiple times consecutively, it throws an error on the browser. Even though you did a great job refactoring the Calculator component to a functional component, it is required that your app is free of error. Kindly add an error handling condition or method. :pray: 

![Screenshot from 2021-09-23 20-48-25](https://user-images.githubusercontent.com/64894927/134530524-fd58c7cb-8d95-4c6f-9a7d-c7cf454bb7a4.png)

Reply:
Thank you for reviewing my project! I didn't do an error handling condition or method because I found the error. The function when the user presses an operation without having entered a number first return nothing was commented.